src/function/table/system/duckdb_extensions.cpp

```diff
@@ -98,16 +102,20 @@ unique_ptr<GlobalTableFunctionState> DuckDBExtensionsInit(ClientContext &context
 	});
 #endif
 	// now check the list of currently loaded extensions
-	auto &loaded_extensions = db.LoadedExtensions();
-	for (auto &ext_name : loaded_extensions) {
+	auto &loaded_extensions = db.LoadedExtensionsData();
```

Comment:
I agree this would be appropriate, but I would feel more comfortable doing that in a separate PR.

I would say goal of this PR is:
1. introducing metadata
2. adding validation on load (that doubles as a check that metadata has to be right)
3. introducing extension_version metadata (when loaded, shown also via `FROM duckdb_extensions()`)

Next one would be for:
1. reading metadata from the local extension_repository, shown also via `FROM duckdb_extensions()`
2. some form of validations on INSTALL (platform + version, or bypassed when allow_extensions_metadata_mismatch is set && unsigned extensions)

One left over question, connected to this: how to propagate version for statically built in extensions? Is that ever relevant?